packages/next-auth/src/providers/qq.js

```diff
@@ -0,0 +1,60 @@
+/** @type {import(".").OAuthProvider} */
+export default function TencentQQ(options) {
+  return {
+    id: "qq",
+    name: "QQ",
+    type: "oauth",
+    authorization: "https://graph.qq.com/oauth2.0/authorize",
+    token: {
+      url: "https://graph.qq.com/oauth2.0/token",
+      async request(context) {
+        const response = await fetch('https://graph.qq.com/oauth2.0/token',{
+          method: 'POST',
+          body: new URLSearchParams({
+            grant_type: "authorization_code",
+            client_id: context.provider.clientId,
+            client_secret: context.provider.clientSecret,
+            code: context.params.code,
+            redirect_uri: context.provider.callbackUrl,
+            fmt: 'json',
+          }),
+        });
+        const tokens = await response.json()
+        return { tokens }
+      },
+    },
+    userinfo: {
+      url: "https://graph.qq.com/oauth2.0/me",
+      async request(context) {
+        const response = await fetch('https://graph.qq.com/oauth2.0/me?' + 
+        new URLSearchParams({
+          access_token: context.tokens.access_token,
+          fmt: 'json'
+        }), {
+          method: 'GET',
+        });
+        const openIDInfo = await response.json();
+        const userInfoResponse = await fetch('https://graph.qq.com/user/get_user_info?' + 
+        new URLSearchParams({
+          access_token: context.tokens.access_token,
+          oauth_consumer_key: openIDInfo.client_id,
+          openid: openIDInfo.openid ,
+        }), {
+          method: 'GET',
+        });
+        return {...await userInfoResponse.json(), openid: openIDInfo.openid};
+      },
+    },
+
```

Comment:
Hey @Tian-que thanks a lot for the new provider!

Small nit: Can you clean up the extra new-line here and on like `57`? 

Otherwise, it looks good to me!

Reply:
> Hey @Tian-que thanks a lot for the new provider!
> 
> Small nit: Can you clean up the extra new-line here and on like `57`?
> 
> Otherwise, it looks good to me!

I did it！o(*￣▽￣*)ブ
Now there are no extra new-line.